tests/components/alarmdecoder/test_config_flow.py

```diff
@@ -0,0 +1,424 @@
+
+    # Zone Number must be int
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={CONF_ZONE_NUMBER: "asd"},
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "zone_select"
+    assert result["errors"] == {CONF_ZONE_NUMBER: "int"}
+
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={CONF_ZONE_NUMBER: zone_number},
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "zone_details"
+
+    # CONF_RELAY_ADDR & CONF_RELAY_CHAN are inclusive
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={**zone_settings, CONF_RELAY_ADDR: "1"},
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "zone_details"
+    assert result["errors"] == {"base": "relay_inclusive"}
+
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={**zone_settings, CONF_RELAY_CHAN: "1"},
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "zone_details"
+    assert result["errors"] == {"base": "relay_inclusive"}
+
+    # CONF_RELAY_ADDR, CONF_RELAY_CHAN must be int
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={**zone_settings, CONF_RELAY_ADDR: "abc", CONF_RELAY_CHAN: "abc"},
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "zone_details"
+    assert result["errors"] == {
+        CONF_RELAY_ADDR: "int",
+        CONF_RELAY_CHAN: "int",
+    }
+
+    # CONF_ZONE_LOOP depends on CONF_ZONE_RFID
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={**zone_settings, CONF_ZONE_LOOP: "1"},
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "zone_details"
+    assert result["errors"] == {CONF_ZONE_LOOP: "loop_rfid"}
+
+    # CONF_ZONE_LOOP must be int
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={**zone_settings, CONF_ZONE_RFID: "rfid123", CONF_ZONE_LOOP: "ab"},
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "zone_details"
+    assert result["errors"] == {CONF_ZONE_LOOP: "int"}
+
+    # CONF_ZONE_LOOP must be between [1,4]
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={**zone_settings, CONF_ZONE_RFID: "rfid123", CONF_ZONE_LOOP: "5"},
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "zone_details"
+    assert result["errors"] == {CONF_ZONE_LOOP: "loop_range"}
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6158285)